urlb879d9e865_ogotalski_elevator_tgz-pJ8-ElevatorTaskJ8/src/by/epam/lab/TransportationTask.java

```diff
@@ -0,0 +1,69 @@
+package by.epam.lab;
+
+import static by.epam.lab.utils.AppLogger.*;
+
+public class TransportationTask implements Runnable {
+	private static final String TRANSPORTATION_INTERRUPDED = "TRANSPORTATION  interrupded ";
+	private static final String WAITING_FOR_DEBOARDING = "WAITING FOR DEBOARDING ";
+	private static final String WANT_TO_DEBOARD = "WANT TO DEBOARD ";
+	private static final String WAITING_FOR_BOARDING = "WAITING FOR BOARDING ";
+	private static final String WANT_TO_BOARD = "WANT TO BOARD ";
+	private static final String READY = "READY ";
+	private ElevatorController controller;
+	private final Passenger passenger;
+
+	public TransportationTask(ElevatorController controller, Passenger passenger) {
+		super();
+		this.controller = controller;
+		this.passenger = passenger;
+	}
+
+	
+
+	@Override
+	public void run() {
+
+		passenger.setTransportationState(TransportationState.IN_PROGRESS);
+
+		try {
+			Floor floor = passenger.getCurrentFloor();
+			Object waitObject = floor.getDispatchStoryContainer();
+			synchronized (controller) {
+				controller.ready();
+				LOG.trace(READY + passenger);
+				controller.notifyAll();
+
+			}
+			synchronized (waitObject) {
+				while (!controller.isWorking())
+					waitObject.wait();
+			}
+
+			synchronized (waitObject) {
+				LOG.trace(WANT_TO_BOARD + passenger.getId());
+				while (!controller.addPassenger(passenger)) {
+					LOG.trace(WAITING_FOR_BOARDING + passenger.getId());
+					waitObject.wait();
+					LOG.trace(WANT_TO_BOARD + passenger.getId());
+				}
+			}
+			floor = passenger.getDestFloor();
+			waitObject = floor.getArrivalStoryContainer();
+			synchronized (waitObject) {
+				LOG.trace(WANT_TO_DEBOARD + passenger.getId());
+				while (!controller.removePassenger(passenger)) {
+					LOG.trace(WAITING_FOR_DEBOARDING + passenger.getId());
+					waitObject.wait();
+					LOG.trace(WANT_TO_DEBOARD + passenger.getId());
+				}
+				passenger.setTransportationState(TransportationState.COMPLETED);
+			}
+
+		} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)